modules/tensor_mechanics/include/materials/ComputeLagrangianStressSmall.h

```diff
@@ -0,0 +1,79 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ComputeLagrangianStressCauchy.h"
+
+/// Provide the Cauchy stress via an objective integration of a small stress
+//
+//    This class provides the Cauchy stress and associated Jacobian through
+//    an objective integration of the small strain constitutive model.
+//
+//    The small strain model implements the computeQpSmallStress()
+//    which must provide the _small_stress and _small_jacobian
+//    properties.
+//
+//    This class is then responsible for completing the cauchy stress
+//    update with an objective integration, providing _cauchy_stress
+//    and _cauchy_jacobian properties
+class ComputeLagrangianStressSmall : public ComputeLagrangianStressCauchy
```

Comment:
I think I understand why you call it `ComputeLagrangianStressSmall` here -- I guess you want to say that the constitutive definition of the objective stress rate can be trivially generalized from the stress constitutive relation under small strain assumptions. But since this class is actually used for large deformations, can we rename it to perhaps `ComputeLagrangianStressRate` or `ComputeLagrangianStressObjective` or something like that?

Reply:
This one I disagree with.  I named the classes by the "thing" the user has to define -- PK1, Cauchy, small stress, or (new) PK2.